### PR TITLE
[build] Replace geometry.hpp/variant with mapbox-base

### PR DESCRIPTION
**Pull request by brunoabinader**
The first iteration of the mapbox-base meta-library.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -103,3 +97,6 @@
 [submodule "vendor/filesystem"]
 	path = vendor/filesystem
 	url = https://github.com/gulrak/filesystem.git
+[submodule "vendor/mapbox-base"]
+	path = vendor/mapbox-base
+	url = https://github.com/mapbox/mapbox-base.git
```
Comment (tmpsantos) [inline .gitmodules]: Mapbox Base will have to add its submodules as HTTPS in order to make it "bot friendly"

**Comment by tmpsantos** [comment]
Btw, the versions of the deps on `mapbox-base` are the same as we use today on gl native, so this should have no impact. Thanks @brunoabinader for starting the `mapbox-base` and organizing better these deps!

File: .gitmodules (hunk 1 above)
Comment (tmpsantos) [reply .gitmodules]: https://github.com/mapbox/mapbox-base/pull/1

File: circle.yml [hunk 2]
```diff
@@ -477,7 +477,9 @@ jobs:
       - install-dependencies: { mason: false, ccache: false }
       - run:
           name: Initialize submodules
-          command: git submodule update --init
+          command: |
+            git submodule update --init
+            git submodule foreach git submodule update --init
```
Comment (pozdnyakov) [inline circle.yml]: git submodule update --init --recursive ?

Comment (brunoabinader) [reply circle.yml]: Unfortunately, there is no way to describe the depth when using `--recursive`. This means that when using that parameter, we end up fetching unnecessary sub-dependencies e.g. some dependencies of ours have `.mason` as subdependency, etc.

Comment (tmpsantos) [reply circle.yml]: With the latest change on `mapbox-base` I don't think this is needed. It will get initialized at configure time.

Comment (brunoabinader) [reply circle.yml]: No - neither `nitpick` nor `android-debug-arm-v7-buck` uses CMake scripts, thus the need for the manual git submodule initialization.

**Pull request merged**